js/script.js

```diff
@@ -1,3 +1,198 @@
-/**
- * Write your code here.
- */
+class Pizza {
+  constructor(name, price, allowedToppings = []) {
+    this.name = name;
+    this.price = price;
+    this.allowedToppings = allowedToppings;
+    this.toppings = [];
+    this.size = null;
+  }
+
+  setSize(size) {
+    this.size = size;
+  }
+
+  addTopping(topping) {
+    if (this.allowedToppings.includes(topping.name)) {
+      this.toppings.push(topping);
+    } else {
+      console.log(`Topping '${topping.name}' is not allowed for ${this.name}.`);
+    }
+  }
+
+  calculateTotalPrice() {
+    let totalPrice = this.price;
+
+    if (this.size === "medium") {
+      totalPrice += 2;
+    } else if (this.size === "large") {
+      totalPrice += 4;
+    }
+
+    totalPrice += this.toppings.reduce(
+      (acc, topping) => acc + topping.price,
+      0
+    );
+
+    return totalPrice;
+  }
+}
+
+class Topping {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+}
+
+const toppingPrices = {
+  avocado: 1,
+  onions: 1,
+  broccoli: 1,
+  zucchini: 1,
+  lobster: 2,
+  salmon: 2,
+  oyster: 2,
+  tuna: 2,
+  bacon: 3,
+  ham: 3,
+  duck: 3,
+  sausage: 3,
+};
+
+const pizzaOptions = [
+  new Pizza("Pizza 1", 8, [
+    "avocado",
+    "broccoli",
+    "onions",
+    "zucchini",
+    "tuna",
+    "ham",
+  ]),
+  new Pizza("Pizza 2", 10, [
+    "broccoli",
+    "onions",
+    "zucchini",
+    "lobster",
+    "oyster",
+    "salmon",
+    "bacon",
+    "ham",
+  ]),
+  new Pizza("Pizza 3", 12, [
+    "broccoli",
+    "onions",
+    "zucchini",
+    "tuna",
+    "bacon",
+    "duck",
+    "ham",
+    "sausage",
+  ]),
+];
+
+function getSelectedPizzaOption() {
+  const pizzaRadios = document.getElementsByName("pizza");
+  for (let i = 0; i < pizzaRadios.length; i++) {
+    if (pizzaRadios[i].checked) {
+      return pizzaOptions[parseInt(pizzaRadios[i].value) - 1];
+    }
+  }
+  return null;
```

Comment:
Give a line break (enter) for each section or part of your code. For example, variable declaration-(enter)-logic scope-(enter)-another logic scope-(enter)-return value.